src/typechecker.erl

```diff
@@ -933,20 +934,15 @@ expand_builtin_aliases(Type) ->
 %% * Remove subtypes of other types in the same union; keeping any() separate
 %% * Merge integer types, including singleton integers and ranges
 %%   1, 1..5, integer(), non_neg_integer(), pos_integer(), neg_integer()
--spec flatten_unions([type()], env(), map()) -> [type()].
-flatten_unions(Tys, Env, Unfolded) ->
-    [ FTy || Ty <- Tys, FTy <- flatten_type(normalize_rec(Ty, Env, Unfolded), Env, Unfolded) ].
+-spec flatten_unions([type()], env()) -> [type()].
+flatten_unions(Tys, Env) ->
+    [ FTy || Ty <- Tys, FTy <- flatten_type(Ty, Env) ].
```

Comment:
I guess we need to update the comment above this function to reflect the changed behaviour. (It is a few lines more than what's possible to select in a code review.)

Btw, flatten unions is still recursive in that it normalizes `A | (B | C)` into a flat union, right? But it doesn't try to normalize the elements in the union nor remove duplicates, is that right?

Reply:
> flatten unions is still recursive in that it normalizes A | (B | C) into a flat union, right?

Yes, I'll add some tests to make that explicit.

> But it doesn't try to normalize the elements in the union nor remove duplicates, is that right?

Exactly.